app/controllers/sensor_readings_controller.rb

```diff
@@ -48,8 +48,51 @@ def fake
     end
   end
 
+  def add
+    if manual_params[:sensor_id].nil? || manual_params[:report_id].nil?
+      respond_to do |format|
+        format.html { redirect_to  root_path }
+        format.js  { render :js => "window.location.href='"+root_path+"'"}
+      end
+      return
+    end
+
+    if manual_params[:timestamp].nil? || manual_params[:calibrated_value].nil? || manual_params[:uncalibrated_value].nil?
+      respond_to do |format|
+        format.html { redirect_to  report_sensor_path(:report_id => manual_params[:report_id] , :id => manual_params[:sensor_id]), notice: reading }
+        format.js  { render :js => "window.location.href='"+report_sensor_path(:report_id => manual_params[:report_id] , :id => manual_params[:sensor_id])+"'"}
+      end
+      return
+    end
+
+    reading = Sensor::Reading.new(
+      sensor_id: manual_params[:sensor_id],
+      smaxtec_timestamp: nil,
+      created_at: manual_params[:timestamp],
+      updated_at: manual_params[:timestamp],
+      calibrated_value: manual_params[:calibrated_value],
+      uncalibrated_value: manual_params[:uncalibrated_value]
+      )
+
+    if reading.save
```

Comment:
`reading.save` will be false if there are validation errors. Be confident in standard Rails validations ;)